consensus/istanbul/validator/weighted.go

```diff
@@ -553,6 +572,36 @@ func (valSet *weightedCouncil) F() int {
 
 func (valSet *weightedCouncil) Policy() istanbul.ProposerPolicy { return valSet.policy }
 
+func (valSet *weightedCouncil) Update(hash common.Hash, blockNum uint64) error {
+	valSet.validatorMu.Lock()
+	defer valSet.validatorMu.Unlock()
+
+	newStakingInfo := reward.GetStakingInfo(blockNum + 1)
```

Comment:
Do you know why this gets the staking info of next block?
```suggestion
	newStakingInfo := reward.GetStakingInfo(blockNum + 1)
```

Reply:
So, we are going to check the staking information for the next block to calculate staked validators.
Like proposer list which is proposer candidates of the next block, validators in the snapshot will be the ones of the next block, not the current block.